[wip] bqn-symbols-doc.el: improve symbol-doc-table generation

AndersonTorres opened the pull request:
Instead of hardcoding the low-level structure of hash table, uses a more typical
list of cons cells as intermediary and convert it afterwards.

Extracted from #22

AndersonTorres committed:
bqn-symbols-doc.el: improve bqn-symbols-doc--symbol-doc-table generation

Instead of hardcoding the low-level structure of hash table, uses a more typical
list of cons cells as intermediary and convert it afterwards.

phikal commented on bqn-symbols-doc.el:
```diff
@@ -25,32 +25,26 @@
 ;; structure. For all intents and purposes, this table should be regarded as
 ;; read-only.
 (defconst bqn-symbols-doc--symbol-doc-table
-  #s(hash-table
```
Why the literal `quote` here?

AndersonTorres replied:
It was too nested I needed to use explicit quote in order to understand.
I will remove this later.

phikal commented on bqn-symbols-doc.el:
```diff
@@ -3194,9 +3016,12 @@ MyError ← {𝕨 \"My custom error\"⊸!⍟(1⊸≢) 𝕩}
+        (ht (make-hash-table :test 'equal)))
+    (dolist (entry table)
+      (puthash (car entry) (cdr entry) ht))
+    ht)
```
It is a waste to recalculate the hash table every time the file is loaded.  By wrapping the entire expression in a `eval-when-compile`, it will be prepared once when installed.

AndersonTorres replied:
Good. I will dedicate this specific optimization to a commit on its own.

AndersonTorres committed:
bqn-symbols-doc.el: remove subr-x dependency

The functionalities of this library were replaced, obviating its necessity.

AndersonTorres committed:
bqn-symbols-doc.el: reword the commentary

AndersonTorres committed:
bqn-symbols-doc.el: eval-when-compile the symbol-doc-table

Givn that it is essentially constant, it can be calculated in advance at
byte-compile time.

AndersonTorres commented:
I believe I can merge this now.
